### ISSUE TITLE
Performance test

### ISSUE THREAD

**Issue by lrq3000** [tqdm/tqdm#51]
Since we aim for tqdm to stay fast compared to the other similar projects, we should maybe make an automatic performance test that should be run everytime we make additions.

I've looked if Travis or another free CI provided performance tests, but it's not currently in the plans:
https://github.com/travis-ci/travis-ci/issues/352
https://github.com/travis-ci/travis-ci/issues/3313

Meanwhile, we could at least make a performance test to run locally and manually by us. There are a few tools to do that:
- [vbench](https://github.com/pydata/vbench) (used by pandas, [introductory post here](http://wesmckinney.com/blog/introducing-vbench-new-code-performance-analysis-and-monitoring-tool/))
- [codespeed](https://github.com/tobami/codespeed) (web app, dunno if there's integration with github, maybe).

So, what do you think guys? Do you know any other alternative? Or do we settle with vbench or codespeed for now to avoid performance regressions?

**Comment by casperdcl** [comment]
Hmm... how about `def test_overhead()` with:

```python
from time import time
from tqdm import trange

tic = time()
for i in trange(int(1e7)):
    pass
time_tqdm = time() - tic

tic = time()
for i in range(int(1e7)):
    pass
time_bench = time() - tic

assert(time_tqdm < 5 * time_bench)
```

**Comment by lrq3000** [comment]
Good idea @casperdcl, it's complicated to reliably calculate the total time performance, but we may reliably compute the overhead compared to range() :)

I'll create a branch with your idea and we'll check if this is stable and reliable enough to merge in the main branch.

**Comment by casperdcl** [comment]
I already did :)

**Comment by lrq3000** [comment]
Ok lol, but then please put the code into a new test file, like "test_perf.py", because I think this kind of test should be separated from the main unit tests (because this is not about testing functionality but performance).

**Comment by lrq3000** [comment]
And please open a PR, I would like to participate :)

**Comment by lrq3000** [comment]
Thank's :)

**Comment by casperdcl** [comment]
Not sure about the merit of putting in a separate file - it'll still have to be called from the current test file...

**Comment by lrq3000** [comment]
It will be called automatically by nose normally, just like the other
tests, no?

2015-11-11 0:37 GMT+01:00 Casper da Costa-Luis <notifications@github.com>:

**Comment by casperdcl** [comment]
ok done :)

**Comment by casperdcl** [comment]
closed by #52
